lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_extract_keys_from_strings_files.rb

```diff
@@ -0,0 +1,119 @@
+          `<SomeIntentDefinitionFile>.strings` from the `Localizable.strings` file that we exported/downloaded back from GlotPress.
+
+          Since we typically merge all `*.strings` original files (e.g. `en.lproj/Localizable.strings` + `en.lproj/InfoPlist.strings` + …)
+          via `ios_merge_strings_file` before sending the originals to translations, we then need to extract the relevant keys and
+          translations back into the `*.lproj/InfoPlist.strings` after we pull those translations back from GlotPress
+          (`ios_download_strings_files_from_glotpress`). This is what this `ios_extract_keys_from_strings_files` action is for.
+        DETAILS
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :source_parent_dir,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_SOURCE_PARENT_DIR',
+                                       description: 'The parent directory containing all the `*.lproj` subdirectories in which the source `.strings` files reside',
+                                       type: String,
+                                       verify_block: proc do |value|
+                                         UI.user_error!("`source_parent_dir` should be a path to an existing directory, but found `#{value}`.") unless File.directory?(value)
+                                         UI.user_error!("`source_parent_dir` should contain at least one `.lproj` subdirectory, but `#{value}` does not contain any.") if Dir.glob('*.lproj', base: value).empty?
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :source_tablename,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_SOURCE_TABLENAME',
+                                       description: 'The basename of the `.strings` file (without the extension) to extract the keys and translations from for each locale',
+                                       type: String,
+                                       default_value: 'Localizable'),
+          FastlaneCore::ConfigItem.new(key: :target_original_files,
+                                       env_name: 'FL_IOS_EXTRACT_KEYS_FROM_STRINGS_FILES_TARGET_ORIGINAL_FILES',
+                                       description: 'The path to the `<base-locale>.lproj/<target-tablename>.strings` file(s) for which we want to extract the keys to. ' \
+                                        + 'Those files, containing the originals (typical `en` or `Base` locale), will be used to determine which keys to extract from the `source_tablename`, ' \
+                                        + 'and the target file path(s) in which to extract the translations will be determined from those path, writing them in the file with the same basename ' \
+                                        +' in each of the other `*.lproj` sibling folders of those target paths',
```

Comment:
> I thought of suggesting to split the base-locale value from the target-tablename one, but I don't think the verify_block below would then be able to read base-locale to check for the file existence. 😞

Also, splitting `target_tablename` and `base_locale` is how it was before the 2nd pass of my work of that PR, but I changed it to that because of the need for this action to support multiple `{target_tablename + target_parent_dir}` pairs (see [this comment above](https://github.com/wordpress-mobile/release-toolkit/pull/338#discussion_r806176480) for details).

---

Because we need to support the case of having both `WordPress/Resources/*/InfoPlist.strings` and `WordPress/WordPressIntents/*/Sites.strings` as target files to generate — but can't really _actually_ expect to use that `*` literal in those paths when passing the values for that parameter at call site — this is why it's difficult to split those into more individual parameters.

One option would have been to split that into `base_locale` and `target_tables`… and make `target_tables` be an Array of Hashes, like:
```
  target_tables: [
    { dir: `WordPress/Resources`, table: `InfoPlist` },
    { dir: `WordPress/WordPressIntents`, table: `Sites` },
  ]
```
or like:
  target_tables: [
    { `WordPress/Resources`: `InfoPlist` },
    { `WordPress/WordPressIntents`: `Sites` },
  ]
```

We couldn't have it be directly a `Hash` (of `{directory => tablename}` values) because nothing tells us that we won't have a case in the future of having to extract to more tables, with 2 target tables ending up being in the same parent dir… but Hashes keys must me unique.

We could then have an `Hash<String, Array<String>>` to do `{ directory => Array<tablenames> }` to support multiple target tables in same parent dir… but at that point that seems even more convoluted than any other option

---

Overall this above has been my reasoning when trying to find the final API, and the train of thoughts which led me to use a simple `Array` like here, with the drawback that it's a tad hard to describe in docs… but I think once you understand the expected values it's quite simple to use.

PS: Note that as an additional bonus, the current solution I ended up using here should theoretically also allow us to have target files that use a different base locale (e.g. `WordPress/Resources/en.lproj/InfoPlist.strings` for the main app but `WordPress/WordPressIntents/en-US.lproj/Sites.strings` or `WordPress/WordPressIntents/Base.lproj/Sites.strings` for the Siri Intents target, or something like that) should we encounter such an odd setup in one of our client apps and targets. That scenario seems unlikely as it's more likely that we try to have the base locales be consistent (including given GlotPress project and thus original locale would be the same)… but we never know what odd beast we might encounter… (also thinking about projects like DayOne and/or PocketCasts which might soon adopt our tooling but might not yet have a straight setup with all edges sharpened properly just yet)

Reply:
Ended up borrowing most of your wording suggestion here, but tweaked the second sentence a bit more, which led to 96a9a1186ba669cd80adda61fbab2a0e7606a9d7 and this description:
```
                                       description: 'The path(s) to the `<base-locale>.lproj/<target-tablename>.strings` file(s) for which we want to extract the keys to. ' \
                                        + 'Each of those files should containing the original strings (typically `en` or `Base` locale) and will be used to determine which keys to extract from the `source_tablename`. ' \
                                        + 'For each of those, the path(s) in which the translations will be extracted will be the files with the same basename in each of the other `*.lproj` sibling folders',
```
I'll merge and go with this but don't hesitate to comment if you have further feedback on that take and I can address them in a follow-up PR.

Reply:
All good 👍